testing/trino-testing-containers/src/main/java/io/trino/testing/containers/BaseTestContainer.java

```diff
@@ -115,7 +116,8 @@ public void start()
                                 "%s initialization failed (attempt %s), will retry. Exception: %s",
                                 this.getClass().getSimpleName(),
                                 event.getAttemptCount(),
-                                event.getLastFailure().getMessage())))
+                                event.getLastFailure().getMessage()))
+                        .abortOn(ContainerLaunchException.class))
```

Comment:
That's a weird choice. Retry... but not so much. Why?

Reply:
Yes that might be too much. I will need to test more.
In case of this container, it filtered out case when app exited with non 0 code. 
There was no chance its restart would help. 
Actually I can't imagine situation where restart would help when app crashes due to some reasons.

Reply:
Droped